src/components/verification-grid/verification-grid.ts

```diff
@@ -0,0 +1,1364 @@
+   * Note: Changing to "History Mode" does not render the history. It is only
+   *       responsible for changing the layout of the verification grid.
+   */
+  private historyMode(decisions: DecisionWrapper[]): void {
+    if (this.decisions.length === 0) {
+      throw new Error("No decisions to show in history");
+    }
+
+    // if the user has completed all their verifications, then all the grid
+    // items would have been removed
+    // therefore, we have to recreate these grid tiles before we can start
+    // rendering history (if they are not present)
+    if (!this.gridTiles || this.gridTiles.length === 0) {
+      this.createSpectrogramElements();
+
+      // TODO: forcing a synchronous update is hacky, and we should definitely remove it
+      this.performUpdate();
+    }
+
+    // TODO: this should only show the touched decision elements for the
+    // grid tiles shown on the current page
+    const touchedDecisionElements = this.touchedDecisionElements();
+    this.showDecisionButtonHighlight(touchedDecisionElements);
+    this.createDecisionHighlight(decisions);
+  }
+
+  /**
+   * @description
+   * Change the verification grid to the "normal mode" layout
+   * Note: Changing to "Verification Mode" does not stop rendering history
+   *       responsible for changing the layout of the verification grid.
+   */
+  private verificationMode(): void {
+    this.removeSubSelection();
+    this.removeDecisionHighlight();
+    this.removeDecisionButtonHighlight();
+    this.historyHead = 0;
+  }
+
+  /**
+   * @description
+   * Changes to the "verification mode" and renders the
+   */
+  private resumeVerification(): void {
+    this.renderCurrentPage();
+    this.verificationMode();
+  }
+
+  private handleNextPageClick(): void {
+    if (this.isViewingHistory()) {
+      this.pageForwardHistory();
+    } else {
+      this.nextPage();
+    }
+  }
+
+  private async nextPage(pagedItems: number = this.gridSize): Promise<void> {
+    this.removeSubSelection();
+    this.removeDecisionHighlight();
+    this.resetSpectrogramSettings();
+
+    if (!this.getPage) {
+      throw new Error("No getPage method found.");
+    }
+
+    const nextPageIndex = this.pagedItems + pagedItems;
+    const nextPage = await this.getPage(nextPageIndex);
+    const nextVerificationModels = nextPage.map(VerificationParser.parse);
+
+    // get all verification models that are not in the undecided tiles
+    const undecidedSubjects = this.undecidedTiles.map((tile: DecisionWrapper) => tile.subject);
+    const decisionSubjects = this.decisions.map((decision: DecisionWrapper) => decision.subject);
+
+    const newVerificationModels = nextVerificationModels.filter(
+      (model: DecisionWrapper) =>
+        !undecidedSubjects.some((subject) => JSON.stringify(model.subject) === JSON.stringify(subject)) &&
+        !decisionSubjects.some((subject) => JSON.stringify(model.subject) === JSON.stringify(subject)),
```

Comment:
Do not do structural equality like this.

We talked about this offline. Don't couple the page sizes for fetching subjects to the size of the grid. Get 25 items in a page and render tiles from that chunk until you run out, and then fetch more.

Doing it this way will obviate your need to look for duplicates from your data source due to the fact that any subject can be decided on (which essentially reorders the list).